toc/rfc/rfc-draft-ipv6-dual-stack-for-cf.md

```diff
@@ -0,0 +1,200 @@
+# Meta
+
+- Name: IPv6 Dual Stack Support for     Cloud Foundry
+- Start Date: 2025-01-22
+- Author(s): @peanball, @a-hassanin, @fmoehler, @dimitardimitrov13, @plamen-bardarov
+- Status: Draft
+- RFC Pull Request: (fill in with PR link after you submit it)
+
+## Summary
+
+IPv6 becomes increasingly prevalent on the internet and slowly also in the networks of large customers.
+
+This RFC proposes to add IPv6 support to various Cloud Foundry components in addition to the existing networking in CF to support IPv4, IPv6, or Dual Stack, i.e. IPv4 *and* IPv6 at the same time. The latter is particularly relevant for basic extension and migration of existing CF foundations.
+
+Furthermore, the goal is to support IPv6 for both ingress and egress traffic, ensuring that Cloud Foundry can handle inbound and outbound network communication using IPv6, further enhancing compatibility and future-proofing the platform.
+
+## Problem
+
+IPv6 support in Cloud Foundry has only reached a few CF components so far. Furthermore, the existing support often only supports one type of networking exclusively.
+
+Additionally, there are areas in the CF design informed by the limited address space of IPv4:
+
+* Workloads in Diego are interconnected via an overlay network managed by the Silk CNI. An overlay network is primarily needed to provide enough address space for the apps running within Diego and offer a network that spans all workloads.
+* On hyperscalers, egress traffic from apps requires a NAT gateway. Providing public IP addresses for all Diego cells is often not economical.
+* BOSH allows assigning a single IP address per BOSH network entry for a VM, not CIDR ranges.
+
+Adding IPv6 support should not transfer those same issues onto the IPv6 implementation.
+
+## Proposal
+
+The goal is to use IPv6 best practices and benefits to the full extent with the following high-level goals:
+
+1. NAT and port forwarding are avoided wherever possible when using IPv6
+2. Application workloads have their own IPv6 address. This allows better correlation between requests and their origin. It is important to note that application workloads shall not be addressable directly from external networks. The ingress and load balancing via Gorouter must remain.
+3. Support of dual stack networking, i.e. IPv4 and IPv6 at the same time. This is primarily for a smoother migration because the IPv6 changes can then be additive.
+4. IPv6 is additive to IPv4 functionality
+
+The ultimate future goal would be to move to pure IPv6 communication but is not a current priority and is not covered in this proposal.
```

Comment:
IPv4?

Reply:
IPv6 networks allow interoperability, e.g. via NAT64 and similar. IPv6-only is still intended as "option" on how to use Cloud Foundry. We will clarify this.

Reply:
Modified that statement to a more realistic goal.